app/src/main/java/com/example/calculatorapp/MainActivity.kt

```diff
@@ -1,15 +1,132 @@
+            }
+            R.id.tvClosingBracket -> {
+                // Implement action for R.id.tvClosingBracket
+                tvDisplay.text = tvDisplay.text.toString() + ")"
+            }
+            R.id.tvDivide -> {
+                // Implement action for R.id.tvDivide
+                tvDisplay.text = tvDisplay.text.toString() + "/"
+            }
+            R.id.tv7 -> {
+                // Implement action for R.id.tv7
+                tvDisplay.text = tvDisplay.text.toString() + "7"
+            }
+            R.id.tv8 -> {
+                // Implement action for R.id.tv8
+                tvDisplay.text = tvDisplay.text.toString() + "8"
+            }
+            R.id.tv9 -> {
+                // Implement action for R.id.tv9
+                tvDisplay.text = tvDisplay.text.toString() + "9"
+            }
+            R.id.tvMultiply -> {
+                // Implement action for R.id.tvMultiply
+                tvDisplay.text = tvDisplay.text.toString() + "*"
+            }
+            R.id.tv4 -> {
+                // Implement action for R.id.tv4
+                tvDisplay.text = tvDisplay.text.toString() + "4"
+            }
+            R.id.tv5 -> {
+                // Implement action for R.id.tv5
+                tvDisplay.text = tvDisplay.text.toString() + "5"
+            }
+            R.id.tv6 -> {
+                // Implement action for R.id.tv6
+                tvDisplay.text = tvDisplay.text.toString() + "6"
+            }
+            R.id.tvSubstract -> {
+                // Implement action for R.id.tvSubstract
+                tvDisplay.text = tvDisplay.text.toString() + "-"
+            }
+            R.id.tv1 -> {
+                // Implement action for R.id.tv1
+                tvDisplay.text = tvDisplay.text.toString() + "1"
+            }
+            R.id.tv2 -> {
+                // Implement action for R.id.tv2
+                tvDisplay.text = tvDisplay.text.toString() + "2"
+            }
+            R.id.tv3 -> {
+                // Implement action for R.id.tv3
+                tvDisplay.text = tvDisplay.text.toString() + "3"
+            }
+            R.id.tvAdd -> {
+                // Implement action for R.id.tvAdd
+                tvDisplay.text = tvDisplay.text.toString() + "+"
+            }
+            R.id.tv0 -> {
+                // Implement action for R.id.tv0
+                tvDisplay.text = tvDisplay.text.toString() + "0"
+            }
+            R.id.tvDot -> {
+                // Implement action for R.id.tvDot
+                tvDisplay.text = tvDisplay.text.toString() + "."
+            }
+            R.id.tvBackspace -> {
+                // Implement action for R.id.tvBackspace
+                if(tvDisplay.text.toString()!="")
+                tvDisplay.text = tvDisplay.text.toString().substring(0,tvDisplay.text.toString().length-1)
+            }
+            R.id.tvEqual -> {
+                // Implement action for R.id.tvEqual
```

Comment:
Since you asked about next steps. This one is interesting. Try creating a function calculate that takes a string parameter and returns the answer for the given expression 🙂

Reply:
Ok I will try...